tests/app/Services/DependencyServiceTest.php

```diff
@@ -5,15 +5,23 @@
 use App\BaseFunctionalTestCase;
 use App\Models\Dependency\Dependency;
 use App\Models\User\User;
+use Auth;
+use Cache;
 use Illuminate\Support\Carbon;
 use Illuminate\Support\Facades\Date;
+use InvalidArgumentException;
 use LogicException;
 
 class DependencyServiceTest extends BaseFunctionalTestCase
 {
     private const GLOBAL_DEPENDENCY = 'DEPENDENCY_ONE';
     private const USER_DEPENDENCY = 'USER_DEPENDENCY_ONE';
 
+    public function foo(): string
```

Comment:
Worth documenting why this method is here (to provide a method to test the resolution upon). Did you consider using a mockery spy instead?

Reply:
Documented :)